### PR TITLE
Added missing admin security annotations

### PR DESCRIPTION
**Pull request by RaimondasSapola**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | In total linter tool revealed 80 unsecured routes. After investigating routes that came with migration remained 17 in total. Tested every route and its behavior after migration and in legacy. See below
| Type?         | bug fix
| Category?     |  BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | ---
| How to test?  | Access to these routes should be denied if no permission are given

## Details

**Final result:**
admin_orders_add_cart_rule| Done!
admin_orders_add_payment| Done!
admin_orders_change_currency| Done!
admin_orders_remove_cart_rule| Done!
admin_orders_resend_email| Done!
admin_orders_update_shipping| Done!
admin_orders_update_invoice_note| Done!
admin_orders_generate_invoice| Legacy didn't require permission. Only required to be enabled in configuration
admin_orders_change_customer_address| Done!
admin_orders_partial_refund| Done!
admin_import_file_upload| Works as is
admin_employees_edit| Done!
admin_employees_toggle_navigation| Doesn’t need permissions
admin_employees_change_form_language| Doesn’t need permissions
admin_currencies_update_live_exchange_rates| Done within action
admin_theme_customize_layouts| Done within action
admin_mail_theme_translate_body| Doesn't work with other roles as intended need to investigate its behavior

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17366)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]
Need a second approve

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 1]
```diff
@@ -564,6 +569,13 @@ public function getInvoicesAction(int $orderId)
     }
 
     /**
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *      redirectRoute="admin_orders_view",
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php]:
```suggestion
     *     redirectRoute="admin_orders_view",
```

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 2]
```diff
@@ -609,6 +621,13 @@ public function updateShippingAction(int $orderId, Request $request): RedirectRe
     }
 
     /**
+     * @AdminSecurity(
+     *     "is_granted('update', 'AdminOrders')",
+     *      redirectRoute="admin_orders_view",
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php]:
```suggestion
     *     redirectRoute="admin_orders_view",
```

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 3]
```diff
@@ -628,6 +647,13 @@ public function removeCartRuleAction(int $orderId, int $orderCartRuleId): Redire
...
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php]:
```suggestion
     *     redirectRoute="admin_orders_view",
```

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 4]
```diff
@@ -771,6 +804,13 @@ public function updateStatusFromListAction(int $orderId, Request $request): Redi
...
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php]:
```suggestion
     *     redirectRoute="admin_orders_view",
```

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 5]
```diff
@@ -935,6 +975,13 @@ public function sendMessageAction(Request $request, int $orderId): Response
...
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php]:
```suggestion
     *     redirectRoute="admin_orders_view",
```

File: src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php [hunk 6]
```diff
@@ -1017,6 +1071,20 @@ public function changeCurrencyAction(int $orderId, Request $request): RedirectRe
         ]);
     }
 
+    /**
+     * @AdminSecurity(
+     *     "is_granted('update', 'AdminOrders')",
+     *      redirectRoute="admin_orders_view",
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php]:
```suggestion
     *     redirectRoute="admin_orders_view",
```

**Comment by matks** [comment]
@RaimondasSapola PrettyCI is not happy 😅

**Comment by RaimondasSapola** [comment]
PrettyCI is happier now :)

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thanks ! going for QA 😊

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by matks** [comment]
@RaimondasSapola Does this PR solve all missing annotations (or will there be another one) ?

**Comment by RaimondasSapola** [comment]
@matks I added annotations for pages migrated by use where security is not handled in the action itself. So if we do not plan to redo other actions to require security annotations, no additional PR's will be required

**Comment by matks** [comment]
Thank you @sarahdib for the hard work on QA
and thanks @RaimondasSapola for the PR ! 🎉

**Pull request merged**